File: code/dotnet/PredictorLibrary/Model2/Outputs/Model2Year.cs
namespace PredictorLibrary.Model2.Outputs;

public class Model2Year
{
    public int CalendarYear { get; set; }
    public int Age { get; set; }
    public int YearIndex { get; set; }
    public decimal PriorYear { get; set; }
    public decimal AmountAtStart { get; set; }

    public decimal InvestmentReturn
    {
        get { return Allocations.Sum(allocation => allocation.InvestmentReturn); }
    }

    public IList<Model2AllocationYear> Allocations { get; set; } = new List<Model2AllocationYear>();

    public decimal AmountAtEnd => AmountAtStart + InvestmentReturn;
}

File: code/angular/src/app/model2/models/Model2Year.ts
export class Model2Year {
    calendarYear!: number;
    age!: number;
    yearIndex!: number;
    priorYear!: number;
    amountAtStart!: number;
    investmentReturn!: number;
    allocations!: Model2AllocationYear[];
    amountAtEnd!: number;
}

export class Model2AllocationYear {
  name!: string;
  amountAtStart!: number;
  investmentReturn!: number;
}
